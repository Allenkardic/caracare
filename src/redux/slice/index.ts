export * from './charactersSlice';
export * from './favouriteCharactersSlice';
export * from './firstEpisodeSlice';
export * from './lastEpisodeSlice';
export * from './singleCharacterSlice';
export * from './settingsSlice';
