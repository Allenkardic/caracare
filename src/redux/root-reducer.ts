import {combineReducers} from '@reduxjs/toolkit';
import {charactersSliceReducer} from './slice';

const rootReducer = combineReducers({
  characters: charactersSliceReducer,
});

export type RootState = ReturnType<typeof rootReducer>;

export default rootReducer;
