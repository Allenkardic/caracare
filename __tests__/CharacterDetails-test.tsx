import 'react-native';
import React from 'react';
import CharacterDetails from '../src/screens/home/characterDetails';

import renderer from 'react-test-renderer';
import {testAppSelector} from '../src/redux/test-app-selector';
import {useAppSelector} from '../src/redux/redux-hooks';

jest.mock('../src/redux/redux-hooks');

const createTestProps = (props: Object) => ({
  ...props,
});

describe('Character Details Screen', () => {
  beforeEach(() => {
    useAppSelector.mockImplementation(testAppSelector);
  });
  afterAll(() => {
    jest.clearAllMocks();
  });
  it('renders correctly', () => {
    let props: any;
    props: createTestProps;
    const mockedParams = {
      route: {
        params: 1,
      },
    };
    const tree = renderer
      .create(<CharacterDetails {...props} {...mockedParams} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
