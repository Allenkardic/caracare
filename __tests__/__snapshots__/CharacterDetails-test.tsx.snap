// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Details Screen renders correctly 1`] = `
<RCTScrollView
  style={
    Object {
      "paddingTop": 24,
    }
  }
>
  <View>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Array [
          Object {
            "marginBottom": 48,
          },
          Object {},
        ]
      }
    >
      <View
        style={
          Object {
            "borderTopLeftRadius": 10,
            "borderTopRightRadius": 10,
            "height": 667,
            "width": "100%",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://rickandmortyapi.com/api/character/avatar/1.jpeg",
            }
          }
          style={
            Array [
              Object {
                "height": "100%",
                "resizeMode": "cover",
                "width": "100%",
              },
              Object {},
            ]
          }
        />
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "space-between",
            "paddingLeft": 24,
            "paddingTop": 24,
          }
        }
      >
        <View
          style={
            Object {
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "rgb(28, 28, 30)",
                  "fontFamily": "Montserrat-Regular",
                  "fontSize": 28,
                },
                Object {
                  "fontSize": 31,
                  "textAlign": "left",
                },
                undefined,
                undefined,
                Object {
                  "fontFamily": "Montserrat-SemiBold",
                },
                undefined,
                undefined,
                undefined,
              ]
            }
          />
          <View
            style={
              Object {
                "flexDirection": "row",
                "justifyContent": "space-between",
                "paddingRight": 24,
              }
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginVertical": 12,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 28,
                    },
                    Object {
                      "fontSize": 25,
                      "textAlign": "left",
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                ,
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 28,
                    },
                    Object {
                      "fontSize": 25,
                      "marginHorizontal": 8,
                      "textAlign": "left",
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#6D6E71",
                  "borderRadius": 10,
                  "justifyContent": "center",
                  "paddingHorizontal": 12,
                  "paddingVertical": 2,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 28,
                    },
                    Object {
                      "fontSize": 25,
                      "textAlign": "left",
                    },
                    Object {
                      "color": "#FFFFFF",
                    },
                    Object {
                      "fontFamily": "Montserrat-Bold",
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              />
            </View>
          </View>
          <View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "marginTop": 24,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 28,
                    },
                    Object {
                      "fontSize": 25,
                      "textAlign": "left",
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                first episode: 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 28,
                    },
                    Object {
                      "fontSize": 25,
                      "textAlign": "left",
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "#6D6E71",
                    "borderRadius": 2.5,
                    "height": 5,
                    "marginHorizontal": 8,
                    "width": 5,
                  }
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 28,
                    },
                    Object {
                      "fontSize": 25,
                      "textAlign": "left",
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              />
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "marginTop": 24,
                  },
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 28,
                    },
                    Object {
                      "fontSize": 25,
                      "textAlign": "left",
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              >
                last episode: 
              </Text>
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 28,
                    },
                    Object {
                      "fontSize": 25,
                      "textAlign": "left",
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              />
              <View
                style={
                  Object {
                    "backgroundColor": "#6D6E71",
                    "borderRadius": 2.5,
                    "height": 5,
                    "marginHorizontal": 8,
                    "width": 5,
                  }
                }
              />
              <Text
                style={
                  Array [
                    Object {
                      "color": "rgb(28, 28, 30)",
                      "fontFamily": "Montserrat-Regular",
                      "fontSize": 28,
                    },
                    Object {
                      "fontSize": 25,
                      "textAlign": "left",
                    },
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                    undefined,
                  ]
                }
              />
            </View>
            <Text
              style={
                Array [
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontFamily": "Montserrat-Regular",
                    "fontSize": 28,
                  },
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "fontSize": 25,
                    "marginTop": 24,
                    "textAlign": "left",
                  },
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                  undefined,
                ]
              }
            >
              total number of episode: 
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
